WalletWasabi.Tests/UnitTests/BranchAndBoundTests.cs

```diff
@@ -0,0 +1,63 @@
+using NBitcoin;
+using System.Collections.Generic;
+using WalletWasabi.BranchNBound;
+using Xunit;
+
+namespace WalletWasabi.Tests.UnitTests
+{
+	public class BranchAndBoundTests
+	{
+		private Random Random { get; } = new();
+
+		[Fact]
+		public void RandomizedTest()
+		{
+			List<Money> utxos = GenList();
+			BranchAndBound selector = new(utxos);
+			Money target = Money.Satoshis(100000000);
+
+			bool successful = selector.TryGetExactMatch(target, out List<Money>? selectedCoins);
+
+			Assert.True(successful);
+			Assert.NotNull(selectedCoins);
+			Assert.Equal(target.Satoshi, CalcEffectiveValue(selectedCoins!).Satoshi);
```

Comment:
Using LINQ, I believe it can be something like:

```suggestion
			Assert.Equal(target.Satoshi, selectedCoins!.Sum().Satoshi);
```